util/circular_buffer.c

```diff
@@ -160,86 +291,26 @@ int32_t cb_size(struct circular_buffer *desc, uint32_t *nb_elements)
  * @brief Write data to the buffer
  * @param desc - Circular buffer reference
  * @param data - Buffer from where data is copied to the circular buffer
- * @param nb_elements - Number of elements to be copied
+ * @param size - Size to write
  * @return
  *  - \ref SUCCESS : On success
  *  - \ref FAILURE : If overflow, if nb_elements == 0 or if desc is NULL
  */
-int32_t cb_write(struct circular_buffer *desc, const void *data,
-		 uint32_t nb_elements)
+int32_t cb_write(struct circular_buffer *desc, const void *data, uint32_t size)
```

Comment:
are these wrappers needed ? I would remove them an just use:
```
cb_operation(desc, (void *)data, size, WRITE_OP);
cb_operation(desc, data, size, READ_OP);

Reply:
This was discussed on cb PR